code/modules/admin/view_variables/debug_variable_appearance.dm

```diff
@@ -0,0 +1,220 @@
+		if(NAMEOF(appearance, mouse_over_pointer))
+			return appearance.mouse_over_pointer
+		if(NAMEOF(appearance, mouse_drag_pointer))
+			return appearance.mouse_drag_pointer
+		if(NAMEOF(appearance, mouse_drop_pointer))
+			return appearance.mouse_drop_pointer
+		if("mouse_drop_zone") // OpenDream didn't implement this yet.
+			return appearance:mouse_drop_zone
+		if(NAMEOF(appearance, mouse_opacity))
+			return appearance.mouse_opacity
+		if(NAMEOF(appearance, name))
+			return appearance.name
+		if(NAMEOF(appearance, opacity))
+			return appearance.opacity
+		if(NAMEOF(appearance, overlays))
+			return appearance.overlays
+		if("override") // only /image has this
+			var/image/image_appearance = appearance
+			return image_appearance.override
+		if(NAMEOF(appearance, pixel_x))
+			return appearance.pixel_x
+		if(NAMEOF(appearance, pixel_y))
+			return appearance.pixel_y
+		if(NAMEOF(appearance, pixel_w))
+			return appearance.pixel_w
+		if(NAMEOF(appearance, pixel_z))
+			return appearance.pixel_z
+		if(NAMEOF(appearance, plane))
+			return appearance.plane
+		if(NAMEOF(appearance, render_source))
+			return appearance.render_source
+		if(NAMEOF(appearance, render_target))
+			return appearance.render_target
+		if(NAMEOF(appearance, suffix))
+			return appearance.suffix
+		if(NAMEOF(appearance, text))
+			return appearance.text
+		if(NAMEOF(appearance, transform))
+			return appearance.transform
+		if(NAMEOF(appearance, underlays))
+			return appearance.underlays
+
+		if(NAMEOF(appearance, parent_type))
+			return appearance.parent_type
+		if(NAMEOF(appearance, type))
+			return "/appearance (as [appearance.type])" // don't fool people
+
+		// These are not documented ones but trackable values. Maybe we'd want these.
+		if(NAMEOF(appearance, animate_movement))
+			return appearance.animate_movement
+		if(NAMEOF(appearance, dir))
+			return appearance.dir
+		if(NAMEOF(appearance, glide_size))
+			return appearance.glide_size
+		if("pixel_step_size")
+			return "" //atom_appearance.pixel_step_size
+			// DM compiler complains this
+
+		// I am not sure if these will be ever detected, but I made a connection just in case.
+		if(NAMEOF(appearance, contents)) // It's not a thing, but I don't believe how DM will change /appearance in future.
+			return appearance.contents
+		if(NAMEOF(appearance, loc)) // same reason above
+			return appearance.loc
+		if(NAMEOF(appearance, vis_contents)) // same reason above
+			return appearance.vis_contents
+		if(NAMEOF(appearance, vis_flags)) // DM document says /appearance has this, but it throws error
+			return appearance.vis_flags
+
+		// we wouldn't need these, but let's these trackable anyway...
+		if(NAMEOF(appearance, density))
+			return appearance.density
+		if(NAMEOF(appearance, screen_loc))
+			return appearance.screen_loc
+		if(NAMEOF(appearance, verbs))
+			return appearance.verbs
+		if(NAMEOF(appearance, tag))
+			return appearance.tag
+		else
+			return RESULT_VARIABLE_NOT_FOUND
```

Comment:
I am not happy with this, but switch() has no way than this doing in proc.

Reply:
replace this with `pass()` and then move this out of the switch chain

Reply:
No point of using pass() here. I just made it out of switch.

Reply:
ah okay